index.bs

```diff
@@ -7529,6 +7575,95 @@ iterable |iterable| and an iterator getter
+        1.  If [=!=] [=HasOwnProperty=](|O|, |key|) is <code>true</code>, then:
+            1.  Let |typedKey| be |key| [=converted to an IDL value=] of type |K|.
+            1.  Let |value| be [=?=] [=Get=](|O|, |key|).
+            1.  Let |typedValue| be |value| [=converted to an IDL value=] of type |V|.
+            1.  Assert: |typedKey| is not yet a key in |result|.
+            1.  Append to |result| a mapping from |typedKey| to |typedValue|.
+    1.  Return |result|.
+</ol>
+
+<p id="open-dictionary-to-es">
+    An IDL <code>{{OpenDictionary}}<…></code> value |D| is [=converted to an
+    ECMAScript value|converted=] to an ECMAScript value as follows:
+</p>
+
+<ol class="algorithm">
+    1.  Let |result| be [=!=] [=ObjectCreate=]([=%ObjectPrototype%=]).
+    1.  Repeat, for each [=OpenDictionary/mapping=] (|key|, |value|) in |D|:
+        1.  Let |esKey| be |key| [=converted to an ECMAScript value=].
+        1.  Let |esValue| be |value| [=converted to an ECMAScript value=].
+        1.  Let |created| be [=!=] [=CreateDataProperty=](|result|, |esKey|, |esValue|).
+        1.  Assert: |created| is <code>true</code>.
+    1.  Return |result|.
+</ol>
+
+<div class="example" id="example-es-open-dictionary">
+    Passing the ECMAScript value <code>{b: 3, a: 4}</code> to a
+    <code>{{OpenDictionary}}&lt;DOMString, double></code> argument
+    would result in the IDL value <code>[ "b" ⇒ 3, "a" ⇒ 4 ]</code>.
+
+    Open dictionaries only consider [=own property|own=] [=enumerable=] properties, so given an IDL operation
+    <code>OpenDictionary&lt;double> identity(OpenDictionary&lt;double> arg)</code> which
+    returns its argument, the following code passes its assertions:
+
+    <pre highlight="js">
+        let proto = {a: 3, b: 4};
+        let obj = {__proto__: proto, d: 5, c: 6}
+        Object.defineProperty(obj, "e", {value: 7, enumerable: false});
+        let result = identity(obj);
+        console.assert(result.a === undefined);
+        console.assert(result.b === undefined);
+        console.assert(result.e === undefined);
+        let entries = Object.entries(result);
+        console.assert(entries[0][0] === "d");
+        console.assert(entries[0][1] === 5);
+        console.assert(entries[1][0] === "c");
+        console.assert(entries[1][1] === 6);
+    </pre>
+
+    Open dictionary keys and values can be constrained, although keys can only be
+    constrained among the three string types.
+    The following conversions have the described results:
+    <table class="data">
+        <thead><th>Value</th><th>Passed to type</th><th>Result</th></thead>
+        <tr>
+            <td><code>{"😞": 1}</code></td>
+            <td><code>{{OpenDictionary}}&lt;ByteString, double></code></td>
+            <td><emu-val>TypeError</emu-val></td>
+        </tr>
+        <tr>
+            <td><code>{"\uD83D": 1}</code></td>
+            <td><code>{{OpenDictionary}}&lt;USVString, double></code></td>
+            <td>[ "\uFFFD" ⇒ 1 ]</td>
```

Comment:
It seems you are inconsistent about notation for the mappings; here you use an arrow, whereas above you use "Repeat, for each mapping (key, value) in D". Both seem reasonable, but it would be good to pick one and maybe mention it in the section that introduces the concept of mappings.